c_src/trans_nif.c

```diff
@@ -89,11 +89,13 @@ derivate(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
   if (count < 1) // can't be empty
     return enif_make_badarg(env);
 
-  if (! (target = (decimal*) enif_make_new_binary(env, (count - 1) * sizeof(decimal), &r)))
+  if (! (target = (decimal*) enif_make_new_binary(env, count * sizeof(decimal), &r)))
     return enif_make_badarg(env); // TODO return propper error
 
+  if (count > 0)
+    target[0] = (decimal) {.coefficient = 0, .exponent = 0, .confidence = 0};
```

Comment:
I think that actually, to keep it consistent it is better keep it 0.

Assuming source array is vs, we can say:
    target[0] = vs[0] - vs[-1]
If we were using normal dalmatiner interpolation, we would copy over from next value:
    vs[-1] = vs[0]
,thus
    target[0] = vs[0] - vs[0] = 0

Don't you think it makes mathematical sense?